### PR TITLE
Compile warning of g_type_class_add_private

### PR DESCRIPTION
**Pull request by yetist**
Avoid compile warning for `g_type_class_add_private`.

### CODE REVIEW

**Comment by raveit65** [comment]
With which gtk3 version you see the warning?

**Comment by yetist** [comment]
glib 2.58

**Comment by sc0w** [comment]
I confirm the warning with git master:

```
mdm-signal-handler.c: In function ‘mdm_signal_handler_class_init’:
mdm-signal-handler.c:465:2: warning: ‘g_type_class_add_private’ is deprecated [-Wdeprecated-declarations]
  g_type_class_add_private(klass, sizeof(MdmSignalHandlerPrivate));
  ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from mdm-signal-handler.c:40:
/usr/include/glib-2.0/gobject/gtype.h:1303:10: note: declared here
 void     g_type_class_add_private       (gpointer                    g_class,
          ^~~~~~~~~~~~~~~~~~~~~~~~
```

and the same warning is in more files.

**Comment by sc0w** [comment]
A lot of more files are pending, but it can be done in more PRs

```
gsm-app.c
gsm-autostart-app.c
gsm-client.c
gsm-xsmp-client.c
gsm-dbus-client.c
gsm-consolekit.c
gsm-systemd.c
gsm-logout-dialog.c
gsm-inhibit-dialog.c
gsm-store.c
gsm-inhibitor.c
gsm-xsmp-server.c
gsm-properties-dialog.c
gsm-app-dialog.c
gsp-app.c
gsp-app-manager.c
```

**Comment by lukefromdc** [comment]
I just ran another test build, very few warnings of any kind now left in the whole package. Not zero but really getting down to it.  Starting a new session worked just fine.

Looking at the code, I see essentially the same change everywhere: the ->priv element is removed from the structure, and the structures themselves are no longer named Private .

File: capplet/gsp-app.c [hunk 1]
```diff
@@ -825,66 +906,67 @@ gsp_app_new (const char   *path,
...
                 g_free (basename);
                 _gsp_app_free_reusable_data (app);
         }
 
-        app->priv->path = g_strdup (path);
+        priv->path = g_strdup (path);
 
-        app->priv->hidden = gsp_key_file_get_boolean (keyfile,
+        priv->hidden = gsp_key_file_get_boolean (keyfile,
                                                       G_KEY_FILE_DESKTOP_KEY_HIDDEN,
```
Comment (raveit65) [inline capplet/gsp-app.c]: Indents, from here ........

File: capplet/gsp-app.c [hunk 2]
```diff
@@ -825,66 +906,67 @@ gsp_app_new (const char   *path,
...
 
-        app->priv->name = gsp_key_file_get_locale_string (keyfile,
+        priv->name = gsp_key_file_get_locale_string (keyfile,
                                                           G_KEY_FILE_DESKTOP_KEY_NAME);
-        app->priv->exec = gsp_key_file_get_string (keyfile,
+        priv->exec = gsp_key_file_get_string (keyfile,
                                                    G_KEY_FILE_DESKTOP_KEY_EXEC);
-        app->priv->comment = gsp_key_file_get_locale_string (keyfile,
+        priv->comment = gsp_key_file_get_locale_string (keyfile,
                                                              G_KEY_FILE_DESKTOP_KEY_COMMENT);
```
Comment (raveit65) [inline capplet/gsp-app.c]: .....to here.

File: capplet/gsp-app.c [hunk 3]
```diff
@@ -825,66 +906,67 @@ gsp_app_new (const char   *path,
...
-                g_free (app->priv->name);
-                app->priv->name = g_strdup (app->priv->exec);
+        if (gsm_util_text_is_blank (priv->name)) {
+                g_free (priv->name);
+                priv->name = g_strdup (priv->exec);
         }
 
-        app->priv->icon = gsp_key_file_get_locale_string (keyfile,
+        priv->icon = gsp_key_file_get_locale_string (keyfile,
                                                           G_KEY_FILE_DESKTOP_KEY_ICON);
```
Comment (raveit65) [inline capplet/gsp-app.c]: Indent.

File: capplet/gsp-app.c [hunk 4]
```diff
@@ -974,35 +1057,36 @@ gsp_app_create (const char *name,
         }
 
         app = g_object_new (GSP_TYPE_APP, NULL);
+        priv = gsp_app_get_instance_private (app);
 
-        app->priv->basename = basename;
-        app->priv->path = g_build_filename (g_get_user_config_dir (),
+        priv->basename = basename;
+        priv->path = g_build_filename (g_get_user_config_dir (),
                                             "autostart",
```
Comment (raveit65) [inline capplet/gsp-app.c]: Intends....

File: mate-session/gsm-autostart-app.c [hunk 5]
```diff
@@ -466,32 +475,32 @@ load_desktop_file (GsmAutostartApp *app)
...
         } else {
-                app->priv->autorestart = FALSE;
+                priv->autorestart = FALSE;
         }
 
-        g_free (app->priv->condition_string);
-        app->priv->condition_string = egg_desktop_file_get_string (app->priv->desktop_file,
+        g_free (priv->condition_string);
+        priv->condition_string = egg_desktop_file_get_string (priv->desktop_file,
                                                                    "AutostartCondition",
```
Comment (raveit65) [inline mate-session/gsm-autostart-app.c]: Indents..

File: mate-session/gsm-autostart-app.c [hunk 6]
```diff
@@ -466,32 +475,32 @@ load_desktop_file (GsmAutostartApp *app)
...
                                                                    NULL);
         setup_condition_monitor (app);
 
         if (phase == GSM_MANAGER_PHASE_APPLICATION) {
                 /* Only accept an autostart delay for the application phase */
-                app->priv->autostart_delay = egg_desktop_file_get_integer (app->priv->desktop_file,
+                priv->autostart_delay = egg_desktop_file_get_integer (priv->desktop_file,
                                                                            GSM_AUTOSTART_APP_DELAY_KEY,
```
Comment (raveit65) [inline mate-session/gsm-autostart-app.c]: Indents

File: mate-session/gsm-autostart-app.c [hunk 7]
```diff
@@ -832,41 +854,43 @@ autostart_app_start_spawn (GsmAutostartApp *app,
...
+                                           EGG_DESKTOP_FILE_LAUNCH_RETURN_STARTUP_ID, &priv->startup_id,
                                            NULL);
         g_free (env[0]);
 
         if (success) {
-                g_debug ("GsmAutostartApp: started pid:%d", app->priv->pid);
-                app->priv->child_watch_id = g_child_watch_add (app->priv->pid,
+                g_debug ("GsmAutostartApp: started pid:%d", priv->pid);
+                priv->child_watch_id = g_child_watch_add (priv->pid,
                                                                (GChildWatchFunc)app_exited,
```
Comment (raveit65) [inline mate-session/gsm-autostart-app.c]: Indents

File: mate-session/gsm-autostart-app.c [hunk 8]
```diff
@@ -926,40 +956,40 @@ autostart_app_start_activate (GsmAutostartApp  *app,
...
         }
 
-        arguments = egg_desktop_file_get_string (app->priv->desktop_file,
+        arguments = egg_desktop_file_get_string (priv->desktop_file,
                                                  GSM_AUTOSTART_APP_DBUS_ARGS_KEY,
                                                  NULL);
 
-        app->priv->proxy = dbus_g_proxy_new_for_name (bus,
+        priv->proxy = dbus_g_proxy_new_for_name (bus,
                                                       name,
```
Comment (raveit65) [inline mate-session/gsm-autostart-app.c]: Indents

File: mate-session/gsm-autostart-app.c [hunk 9]
```diff
@@ -926,40 +956,40 @@ autostart_app_start_activate (GsmAutostartApp  *app,
...
                 g_set_error (error,
                              GSM_APP_ERROR,
                              GSM_APP_ERROR_START,
                              "Unable to start application: unable to create proxy for client");
                 return FALSE;
         }
 
-        app->priv->proxy_call = dbus_g_proxy_begin_call (app->priv->proxy,
+        priv->proxy_call = dbus_g_proxy_begin_call (priv->proxy,
                                                          "Start",
```
Comment (raveit65) [inline mate-session/gsm-autostart-app.c]: Indents

File: mate-session/gsm-consolekit.c [hunk 10]
```diff
@@ -177,64 +176,66 @@ gsm_consolekit_ensure_ck_connection (GsmConsolekit  *manager,
...
         connection_error = NULL;
+        priv = gsm_consolekit_get_instance_private (manager);
 
-        if (manager->priv->dbus_connection == NULL) {
+        if (priv->dbus_connection == NULL) {
                 DBusConnection *connection;
 
-                manager->priv->dbus_connection = dbus_g_bus_get (DBUS_BUS_SYSTEM,
+                priv->dbus_connection = dbus_g_bus_get (DBUS_BUS_SYSTEM,
                                                                  &connection_error);
```
Comment (raveit65) [inline mate-session/gsm-consolekit.c]: Indents

File: mate-session/gsm-inhibit-dialog.c [hunk 11]
```diff
@@ -901,17 +900,17 @@ setup_dialog (GsmInhibitDialog *dialog)
                           G_CALLBACK (on_response),
                           dialog);
 
-        dialog->priv->list_store = gtk_list_store_new (NUMBER_OF_COLUMNS,
+        dialog->list_store = gtk_list_store_new (NUMBER_OF_COLUMNS,
                                                        GDK_TYPE_PIXBUF,
```
Comment (raveit65) [inline mate-session/gsm-inhibit-dialog.c]: Indents

File: mate-session/gsm-systemd.c [hunk 12]
```diff
@@ -177,64 +175,67 @@ gsm_systemd_ensure_sd_connection (GsmSystemd  *manager,
...
 
-    if (manager->priv->dbus_connection == NULL) {
+    priv = gsm_systemd_get_instance_private (manager);
+
+    if (priv->dbus_connection == NULL) {
         DBusConnection *connection;
 
-        manager->priv->dbus_connection = dbus_g_bus_get (DBUS_BUS_SYSTEM,
+        priv->dbus_connection = dbus_g_bus_get (DBUS_BUS_SYSTEM,
                                                          &connection_error);
```
Comment (raveit65) [inline mate-session/gsm-systemd.c]: Indents

File: mate-session/gsm-systemd.c [hunk 13]
```diff
@@ -746,7 +763,7 @@ gsm_systemd_can_hibernate (GsmSystemd *manager)
     return FALSE;
   }
   
-  res = dbus_g_proxy_call_with_timeout (manager->priv->sd_proxy,
+  res = dbus_g_proxy_call_with_timeout (priv->sd_proxy,
 					"CanHibernate",
```
Comment (raveit65) [inline mate-session/gsm-systemd.c]: Indents

File: mate-session/gsm-systemd.c [hunk 14]
```diff
@@ -783,7 +802,7 @@ gsm_systemd_can_suspend (GsmSystemd *manager)
     return FALSE;
   }
   
-  res = dbus_g_proxy_call_with_timeout (manager->priv->sd_proxy,
+  res = dbus_g_proxy_call_with_timeout (priv->sd_proxy,
 					"CanSuspend",
```
Comment (raveit65) [inline mate-session/gsm-systemd.c]: Indents

File: mate-session/gsm-systemd.c [hunk 15]
```diff
@@ -818,7 +839,7 @@ gsm_systemd_attempt_hibernate (GsmSystemd *manager)
     return;
   }
   
-  res = dbus_g_proxy_call_with_timeout (manager->priv->sd_proxy,
+  res = dbus_g_proxy_call_with_timeout (priv->sd_proxy,
 					"Hibernate",
```
Comment (raveit65) [inline mate-session/gsm-systemd.c]: Indents

File: mate-session/gsm-systemd.c [hunk 16]
```diff
@@ -849,7 +872,7 @@ gsm_systemd_attempt_suspend (GsmSystemd *manager)
     return;
   }
   
-  res = dbus_g_proxy_call_with_timeout (manager->priv->sd_proxy,
+  res = dbus_g_proxy_call_with_timeout (priv->sd_proxy,
 					"Suspend",
```
Comment (raveit65) [inline mate-session/gsm-systemd.c]: Indents

**Comment by raveit65** [comment]
What a massive commit. I will start testing now.

File: mate-session/gs-idle-monitor.c [hunk 17]
```diff
@@ -390,14 +386,12 @@ idle_monitor_watch_free (GSIdleMonitorWatch *watch)
 static void
 gs_idle_monitor_init (GSIdleMonitor *monitor)
 {
-        monitor->priv = GS_IDLE_MONITOR_GET_PRIVATE (monitor);
-
-        monitor->priv->watches = g_hash_table_new_full (NULL,
+        monitor->watches = g_hash_table_new_full (NULL,
                                                         NULL,
```
Comment (raveit65) [inline mate-session/gs-idle-monitor.c]: Indents

File: mate-session/gsm-manager.c [hunk 18]
```diff
@@ -1889,34 +1973,37 @@ static gboolean
...
                 if (error != NULL) {
                         g_critical ("error getting session bus: %s", error->message);
                         g_error_free (error);
                 }
                 exit (1);
         }
 
-        manager->priv->bus_proxy = dbus_g_proxy_new_for_name (manager->priv->connection,
+        priv->bus_proxy = dbus_g_proxy_new_for_name (priv->connection,
                                                               DBUS_SERVICE_DBUS,
```
Comment (raveit65) [inline mate-session/gsm-manager.c]: Indents

File: mate-session/gsm-manager.c [hunk 19]
```diff
@@ -2137,37 +2243,37 @@ _handle_client_end_session_response (GsmManager *manager,
...
+        } else if (priv->phase == GSM_MANAGER_PHASE_END_SESSION) {
                 if (do_last) {
                         /* This only makes sense if we're in part 1 of
                          * GSM_MANAGER_PHASE_END_SESSION. Doing this in part 2
                          * can only happen because of a buggy client that loops
                          * wanting to be last again and again. The phase
                          * timeout will take care of this issue. */
-                        manager->priv->next_query_clients = g_slist_prepend (manager->priv->next_query_clients,
+                        priv->next_query_clients = g_slist_prepend (priv->next_query_clients,
                                                                              client);
```
Comment (raveit65) [inline mate-session/gsm-manager.c]: i am getting bored....indents.

File: mate-session/gsm-manager.c [hunk 20]
```diff
@@ -3017,33 +3168,38 @@ request_shutdown (GsmManager *manager)
...
                 g_debug ("GsmManager: inhibit dialog already up");
-                gtk_window_present (GTK_WINDOW (manager->priv->inhibit_dialog));
+                gtk_window_present (GTK_WINDOW (priv->inhibit_dialog));
                 return;
         }
 
-        manager->priv->inhibit_dialog = gsm_inhibit_dialog_new (manager->priv->inhibitors,
-                                                                manager->priv->clients,
+        priv->inhibit_dialog = gsm_inhibit_dialog_new (priv->inhibitors,
+                                                                priv->clients,
```
Comment (raveit65) [inline mate-session/gsm-manager.c]: Again...Indents

File: mate-session/gsm-manager.c [hunk 21]
```diff
@@ -3052,39 +3208,46 @@ request_hibernate (GsmManager *manager)
...
```
Comment (raveit65) [inline mate-session/gsm-manager.c]: Why?   Indents

File: mate-session/gsm-manager.c [hunk 22]
```diff
@@ -3099,32 +3262,36 @@ request_switch_user (GsmManager *manager)
...
```
Comment (raveit65) [inline mate-session/gsm-manager.c]: Common....Indents...

File: mate-session/gsm-presence.c [hunk 23]
```diff
@@ -153,24 +152,27 @@ on_idle_timeout (GSIdleMonitor *monitor,
...
-            && presence->priv->idle_timeout > 0) {
+        if (! priv->screensaver_active
+            && priv->idle_enabled
+            && priv->idle_timeout > 0) {
                 g_debug ("GsmPresence: adding idle watch");
 
-                presence->priv->idle_watch_id = gs_idle_monitor_add_watch (presence->priv->idle_monitor,
-                                                                           presence->priv->idle_timeout,
+                priv->idle_watch_id = gs_idle_monitor_add_watch (priv->idle_monitor,
+                                                                           priv->idle_timeout,
```
Comment (raveit65) [inline mate-session/gsm-presence.c]: Indents  :/

File: mate-session/gsm-presence.c [hunk 24]
```diff
@@ -224,21 +235,21 @@ on_bus_name_owner_changed (DBusGProxy  *bus_proxy,
                    && strlen (new_service_name) > 0) {
                 /* service added */
                 error = NULL;
-                presence->priv->screensaver_proxy = dbus_g_proxy_new_for_name_owner (presence->priv->bus_connection,
+                priv->screensaver_proxy = dbus_g_proxy_new_for_name_owner (priv->bus_connection,
                                                                                      GS_NAME,
```
Comment (raveit65) [inline mate-session/gsm-presence.c]: again...always the same!   Indents

File: mate-session/gsm-presence.c [hunk 25]
```diff
@@ -277,28 +291,30 @@ gsm_presence_constructor (GType                  type,
...
+        priv = gsm_presence_get_instance_private (presence);
 
         res = register_presence (presence);
         if (! res) {
                 g_warning ("Unable to register presence with session bus");
         }
 
-        presence->priv->bus_proxy = dbus_g_proxy_new_for_name (presence->priv->bus_connection,
+        priv->bus_proxy = dbus_g_proxy_new_for_name (priv->bus_connection,
                                                                DBUS_SERVICE_DBUS,
```
Comment (raveit65) [inline mate-session/gsm-presence.c]: omg......Indents

**Comment by raveit65** [comment]
Why not using the `git rebase -i <last-commit-id-before-PR> for editing the commits?
Much cleaner and professional as adding stupid commits for a indents-fix.

**Comment by raveit65** [comment]
No, you can also edit a existing commit with the rebase command.
Choose `edit` in vi editor.

**Comment by lukefromdc** [comment]
I have never used vi effectively and thus would have to work on the files separately

File: capplet/gsm-app-dialog.c [hunk 26]
```diff
@@ -414,30 +411,12 @@ gsm_app_dialog_class_init (GsmAppDialogClass *klass)
                                                               "comment",
                                                               NULL,
                                                               G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
-
-        g_type_class_add_private (klass, sizeof (GsmAppDialogPrivate));
 }
 
 static void
 gsm_app_dialog_init (GsmAppDialog *dialog)
 {
```
Comment (sc0w) [inline capplet/gsm-app-dialog.c]: this function is empty, it can't be removed?

Comment (yetist) [reply capplet/gsm-app-dialog.c]: This function is required for gobject system, it is instance initial function, called by other function.

**Comment by raveit65** [comment]
Thanks for fixing indents.
Well, last commit we can squash into commit before.
Sadly, one big commit for fixing indents will be left. No chance to squashing them in other commits before :/
Here is a TODO how to use the git rebase command for editing a commit for future PRs.
http://www.raphaelfabeni.com.br/git-editing-commits-part-2/

**Comment by raveit65** [comment]
@sc0w @lukefromdc 
Maybe we squash all in one commit?
It's already one huge commit there.

**Comment by lukefromdc** [comment]
I also agree it is best to squash this

**Comment by raveit65** [comment]
I was able to split the indents commit in several pieces in result i could squash them in previous commits.
For references how this works https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits
Merged with 6 commits:
https://github.com/mate-desktop/mate-session-manager/commit/851d9fedf02a501c8567ab03f35a0cabd7b4a2d1
https://github.com/mate-desktop/mate-session-manager/commit/582bcbad9587c34247689947f098647977545ba1
https://github.com/mate-desktop/mate-session-manager/commit/6934f934da478bb23fc85cf83d8ba5d620ec7995
https://github.com/mate-desktop/mate-session-manager/commit/f3e8cce5adf6a73c9eb44885014b6872979f4704
https://github.com/mate-desktop/mate-session-manager/commit/bdec713320b7487c429bd4415fdde55cf48cb44b
https://github.com/mate-desktop/mate-session-manager/commit/6b7875b9714d5872f80b6b4012f24fb8f548aca2

@yetist 
Thanks for your your hard work.....
........ hopefully i don't have to do this again for you ;)

**Pull request closed without merging**